Firestore/Source/Model/FSTFieldValue.h

```diff
@@ -85,6 +86,12 @@ enum class ServerTimestampBehavior { None, Estimate, Previous };
  */
 @interface FSTFieldValue<__covariant T> : NSObject
 
+/**
+ * Returns the 'type' of this FSTFieldValue. Used for RTTI (rather than isKindOfClass)
+ * to ease migration to C++.
+ */
+- (firebase::firestore::model::FieldValue::Type)type;
```

Comment:
Feel free to insert a `using firebase::firestore::model::FieldValue` in here to keep usage concise.

Also, FYI if you declared this as

```objc
@property(nonatomic, assign, readonly) FieldValue::Type type;
```

And then added `@dynamic type;` to the `@implementation` your existing implementation would look like a property that you could access as `value.type` (without creating a useless ivar).

This isn't going to be around for long so I don't insist at all.

Reply:
Done. However, I roughly modelled the 'type' parameter after 'typeOrder', so I assume the same should be done to typeOrder. It turns out, typeOrder was already being used as as value.type, but lacked the `@dynamic`. So I added that. PTAL to ensure that makes sense.